### PR TITLE
.editorconfig: File added

### LINKED ISSUE #4097: Add .editorconfig file
Our style follows the example given on http://editorconfig.org/ for Python and YAML
difficulty/newcomer

### PR DESCRIPTION
**Pull request by aryanshar**
Closes #4097

Original indented, and thus didnt link:

              Closes #4097 

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by palash25** [comment]
The following test is failing:
`tests/misc/ContextManagersTest.py`
Fix the assertion error.

**Comment by palash25** [comment]
Even though gitmate ignores it I think the shortog of your commit message should have the files name first. Maybe something like `.editorconfig: File added`.

**Comment by aryanshar** [comment]
@palash25, Updated ! Please Review.

**Comment by aryanshar** [comment]
@palash25 No issues and thanks for pointing where i was doing the error. I've update it now!

**Comment by aryanshar** [comment]
@palash25  Rebased and updated !!

**Comment by palash25** [comment]
ack c181d5c

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# top-most EditorConfig file
+root = true
+
+[*]
+end_of_line = lf
+insert_final_newline = true
+max_line_length = 80
+
+# Set default charset and indentation for python files
+[{*.py, coalib/**/*.py, ./coala}]
```
Comment (aashutoshrathi) [inline .editorconfig]: Please remove the spaces after `,`.
Check http://editorconfig.org/ for more references.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+charset = utf-8
+indent_style = space
+indent_size = 4
+trim_trailing_whitespace = true
+
+# 2 space indentation for yml fies and package.json
+[{package.json,*.yml}]
+indent_style = space
+indent_size = 2
+trim_trailing_whitespace = true
```
Comment (newbazz) [inline .editorconfig]: Add a new line at the end of file.

**Comment by RaiVaibhav** [comment]
Maintainers close this PR as issue is already assigned to someone else

**Comment by newbazz** [comment]
@RaiVaibhav thats exactly what I was thinking.. :P

**Comment by jayvdb** [comment]
Already done.

**Pull request closed without merging**